changelogs/fragments/9238-locale-gen-rewrite.yml

```diff
@@ -0,0 +1,13 @@
+minor_changes:
+  - "locale_gen - invert the logic to determine ``ubuntu_mode``, making it look first for ``/etc/locale.gen`` (set ``ubuntu_mode`` to ``False``)
+    and only then looking for ``/var/lib/locales/supported.d/`` (set ``ubuntu_mode`` to ``True``)
+    (https://github.com/ansible-collections/community.general/pull/9238,
+    https://github.com/ansible-collections/community.general/issues/9131,
+    https://github.com/ansible-collections/community.general/issues/8487)."
+  - >
+    locale_gen - new return value ``mechanism`` to better express the semantics of the ``ubuntu_mode``, with the possible values being either
+    ``glibc`` (``ubuntu_mode=False``) or ``ubuntu_legacy`` (``ubuntu_mode=True``) (https://github.com/ansible-collections/community.general/pull/9238).
+deprecated_features:
+  - >
+    locale_gen - ``ubuntu_mode=True``, or ``mechanism=ubuntu_legacy`` is deprecated and will be removed in community.general 13.0.0
+    (https://github.com/ansible-collections/community.general/pull/9238).
```

Comment:
Shouldn't we call `module.deprecate()` in case that mechanism is used?

Reply:
Mixed feelings.
1. All new systems use the glibc model as far as we can tell. 
2. If an user running a new, supported Ubuntu/Debian Linux falls into that mode, then it is likely something wrong in their installation, and there is nothing the module can do to help, no feature flag, no switches
3. If they are running on old systems, they'll know it, and we'll only be telling them their old system is old.
4. That mode possibly doesn't even work, in which case they should be able to quickly detect and work around the issue. 

I don't see much benefit in generating an actual depreciation message there. If pushing the removal date later in the future helps mitigate your concerns, I'm happy to do it.

Reply:
We still are deprecating something and have to show a deprecation message to affected users. It doesn't matter whether we remove the feature in the next major release or in 5 years, we still have to show that message.

Reply:
Fair enough